internal/server/iigointernal/legislature.go

```diff
@@ -0,0 +1,166 @@
+		return voting.BallotBox{}
+	}
+	l.budget -= serviceCharge
+	ruleVote := voting.RuleVote{}
+
+	//TODO: check if rule is valid, otherwise return empty ballot, raise error?
+	ruleVote.SetRule(ruleID)
+
+	//TODO: intersection of islands alive and islands chosen to vote in case of client error
+	//TODO: check if remaining slice is >0, otherwise return empty ballot, raise error?
+	ruleVote.SetVotingIslands(clientIDs)
+
+	ruleVote.GatherBallots(iigoClients)
+	//TODO: log of vote occurring with ruleID, clientIDs
+	//TODO: log of clientIDs vs islandsAllowedToVote
+	//TODO: log of ruleID vs s.RuleToVote
+	return ruleVote.GetBallotBox()
+}
+
+//Speaker declares a result of a vote (see spec to see conditions on what this means for a rule-abiding speaker)
+//Called by orchestration
+func (l *legislature) announceVotingResult() error {
+
+	rule, result, err := l.clientSpeaker.DecideAnnouncement(l.ruleToVote, l.votingResult)
+
+	if err == nil {
+		//Deduct action cost
+		l.budget -= serviceCharge
+
+		//Reset
+		l.ruleToVote = ""
+		l.votingResult = false
+
+		//Perform announcement
+		broadcastToAllIslands(shared.TeamIDs[l.SpeakerID], generateVotingResultMessage(rule, result))
+		return l.updateRules(rule, result)
+	}
+	return nil
+}
+
+func generateVotingResultMessage(ruleID string, result bool) map[int]baseclient.Communication {
+	returnMap := map[int]baseclient.Communication{}
+
+	returnMap[RuleName] = baseclient.Communication{
+		T:        baseclient.CommunicationString,
+		TextData: ruleID,
+	}
+	returnMap[RuleVoteResult] = baseclient.Communication{
+		T:           baseclient.CommunicationBool,
+		BooleanData: result,
+	}
+
+	return returnMap
+}
+
+//reset resets internal variables for safety
+func (l *legislature) reset() {
+	l.ruleToVote = ""
+	l.ballotBox = voting.BallotBox{}
+	l.votingResult = false
+}
+
+// updateRules updates the rules in play according to the result of a vote.
+func (l *legislature) updateRules(ruleName string, ruleVotedIn bool) error {
+	l.budget -= serviceCharge
+	//TODO: might want to log the errors as normal messages rather than completely ignoring them? But then Speaker needs access to client's logger
+	notInRulesCache := errors.Errorf("Rule '%v' is not available in rules cache", ruleName)
+	if ruleVotedIn {
+		// _ = rules.PullRuleIntoPlay(ruleName)
+		err := rules.PullRuleIntoPlay(ruleName)
+		if err != nil {
+			if err.Error() == notInRulesCache.Error() {
+				return err
+			}
+		}
+	} else {
+		// _ = rules.PullRuleOutOfPlay(ruleName)
+		err := rules.PullRuleOutOfPlay(ruleName)
+		if err != nil {
+			if err.Error() == notInRulesCache.Error() {
+				return err
+			}
+		}
+
+	}
+	return nil
+
+}
+
+func (l *legislature) appointNextJudge(clientIDs []shared.ClientID) shared.ClientID {
+	l.budget -= serviceCharge
+	var election voting.Election
+	election.ProposeElection(baseclient.Judge, voting.Plurality)
+	election.OpenBallot(clientIDs)
+	election.Vote(iigoClients)
+	return election.CloseBallot()
```

Comment:
is it fair to charge the speaker for this if they have no say in doing it. More of a design point, the code looks fine otherwise

Reply:
For MVP this is sufficient. We have plans to change this to allow the roles to choose whether to perform an action and then be charged only if they performed it.